docker-files-for-Gitlab-CI-rust/rustup/Dockerfile

```diff
@@ -1,66 +1,46 @@
 
 # install tools and dependencies
 RUN apt-get -y update && \
-  apt-get install -y --no-install-recommends \
-	software-properties-common curl git file binutils binutils-dev snapcraft \
-  make cmake ca-certificates g++ zip dpkg-dev python rhash rpm openssl gettext \
-  build-essential clang libclang-dev pkg-config libssl-dev libudev-dev ruby-dev time
-
-#install nodejs and yarn
-RUN curl -sL https://deb.nodesource.com/setup_10.x | bash - 
-RUN curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add -
-RUN echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list
-RUN apt-get -y update && apt-get install -y --no-install-recommends \ 
-  nodejs yarn
-
-# install java8
-RUN add-apt-repository -y ppa:webupd8team/java
-RUN apt update
-RUN echo oracle-java8-installer shared/accepted-oracle-license-v1-1 select true | sudo /usr/bin/debconf-set-selections
-RUN apt install -y oracle-java8-installer
-
-# install AWS CLI
-RUN curl -O https://bootstrap.pypa.io/get-pip.py && \
-    python get-pip.py && pip install awscli
-
-# install FPM
-RUN gem install fpm
-
-# install rustup
-RUN curl https://sh.rustup.rs -sSf | sh -s -- -y
+	apt-get install -y --no-install-recommends \
+	software-properties-common curl git \
+	make cmake ca-certificates g++ rhash \
+	gcc pkg-config libudev-dev time
 
-# rustup directory
-ENV PATH /root/.cargo/bin:$PATH
+# removed:
+# binutils binutils-dev snapcraft gettext file python build-essential zip dpkg-dev rpm libssl-dev openssl ruby-dev
 
-# setup rust beta and nightly channel's
-RUN rustup install beta
-RUN rustup install nightly
-
-# install wasm toolchain for polkadot
-RUN rustup target add wasm32-unknown-unknown --toolchain nightly
-# Install wasm-gc. It's useful for stripping slimming down wasm binaries. 
-# (polkadot)
-RUN cargo +nightly install --git https://github.com/alexcrichton/wasm-gc
+#install nodejs and yarn
+RUN curl -sL https://deb.nodesource.com/setup_10.x | bash - && \
+	curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - && \
+	echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list && \
+	apt-get -y update && apt-get install -y --no-install-recommends \
+	nodejs yarn
 
-# setup default stable channel
-RUN rustup default stable
+RUN mkdir /parity-ethereum
+WORKDIR /parity-ethereum
 
-# show backtraces
-ENV RUST_BACKTRACE 1
+RUN cargo install cargo-audit sccache
 
 # cleanup
-RUN apt autoremove -y
-RUN apt clean -y
-RUN rm -rf /tmp/* /var/tmp/*
+RUN echo cleanup && \
+	apt-get autoremove -y && \
+	apt-get clean -y && \
+	rm -rf /tmp/* /var/tmp/*
 
-#compiler ENV
+# compiler ENV
 ENV CC gcc
 ENV CXX g++
+ENV CARGO_TARGET x86_64-unknown-linux-gnu
+ENV CARGO_HOME /parity-ethereum/.cargo/
+# cache handler
+ENV RUSTC_WRAPPER sccache
+# show backtraces
+ENV RUST_BACKTRACE 1
+
+# pre-download repositories
+RUN git clone https://github.com/paritytech/parity-ethereum.git . && \
```

Comment:
Does this bring any noticeable speedup actually? How much?
I probably understand the rationale, but this just rubs me the wrong way.

Reply:
in CI config there are 
```
  GIT_STRATEGY:                    fetch
  GIT_SUBMODULE_STRATEGY:          recursive
```
settings now, it will help to economy ~ a minute on each pipeline, submodules really take time to clone.